### PR TITLE
[sw/vendor] Pull OT-specific C code out of sw/vendor/riscv_compliance

### PR DESCRIPTION
**Pull request by mcy**
This change makes it so that all OT-specific code that needs to run to make the compliance tests work is outside of the vendor boundary, making it easier to make changes to OT libraries without having to do a vendor patch update in lockstep (which makes diffs quite messy).

### CODE REVIEW

**Review by imphil — COMMENTED** [review]
Thanks for this cleanup @mcy, that's nicer indeed. Unfortunately, the timing was slightly unfortunate with #3494 getting in merged just a bit before this PR came in, and your rebase didn't fully work out (there are still merge markers in the patches).

I'd suggest this approach:

* Make sure you're basing your work on the current master of OT. There have been small changes to the vendor tool that omit the signature lines and diffstat.
* Clone our fork of the RISC-V compliance repo at https://github.com/lowrisc/riscv-compliance
* Switch to the 'ot' branch, and modify the files there. Fold your changes (rebase) into the second commit into the series ("Add OpenTitan target")
* Run the vendor tool in OpenTitan and override the patch repo to your local one, e.g.

```
./util/vendor.py sw/vendor/riscv_compliance.vendor.hjson --desc-override patch_repo.url=$HOME/src/riscv-compliance --refresh-patches
```

This should give you an updated tree and updated patches.

After this PR is reviewed, force-push the `ot` branch in riscv-compliance to https://github.com/lowrisc/riscv-compliance. (No need for a PR in the riscv-compliance repo.)

File: sw/device/riscv_compliance_support/support.c [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    base_printf("SIG: %08x\r\n", begin_signature[i]);
+  }
+
+  base_printf("PASS!\r\n");
+
+  // The "End" string here is a workaround to pytest console parsing.
+  // Without additional characters, the "\n" from above is not always
+  // detected, and this causes pytest to register the test as a false failure.
+  // This needs to be debugged further to see if it's a setup or hw issue.
+  base_printf("End");
```
[the remark this answers is not in the document]
Comment (mcy) [reply sw/device/riscv_compliance_support/support.c:32]: Let's just do that in a followup. This PR is blocking another change of mine.

**Comment by mcy** [comment]
That command line fixed things, but it really aught to be documented. Every time I've had to use the vendor tool I've broken something in the process of trying to change things. Maybe we should just add a command that picks up changes in a vendor directory and generates a patch for them?

File: sw/device/riscv_compliance_support/support.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (imphil) [reply sw/device/riscv_compliance_support/support.c:32]: This is trivial and regresses existing behavior, so please just get it into this PR. It shouldn't delay getting this PR merged by more than 10 minutes.

File: sw/device/riscv_compliance_support/support.h [hunk 2]
```diff
@@ -0,0 +1,19 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+/**
+ * \brief OpenTitan-specific wrapper for RISC-V Compliance
```
Comment (lenary) [inline sw/device/riscv_compliance_support/support.h:6]:
```suggestion
 * @brief OpenTitan-specific wrapper for RISC-V Compliance
```

File: sw/device/riscv_compliance_support/support.h [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+ * This header provides a single function, `ot_compliance_main`, which is
+ * intentionally outside of the `sw/vendor` directory, so changes in OpenTitan
+ * libraries do not require generating patches for `riscv-compliance`.
+ */
+
+// NOTE: **DO NOT** #include any files here, since that will leak OpenTitan
+// library interfaces past the vendoring boundary.
+
+/**
+ * Main function for RISC-V Compliance, provided as a out-of-band.
```
Comment (lenary) [inline sw/device/riscv_compliance_support/support.h:17]:
```suggestion
 * Main function for RISC-V Compliance, provided as an out-of-band.
```

As an out-of-band what, this comment seems incomplete.

File: sw/device/riscv_compliance_support/support.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (lenary) [reply sw/device/riscv_compliance_support/support.c:32]: What is this change blocking? The uart.h changes?

**Review by lenary — APPROVED** [review]
LGTM, but I would like you to address
- Philipp's comments
- Please check for typos in your comments. I've caught a few, so there may be more lurking.

File: sw/device/riscv_compliance_support/support.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (mcy) [reply sw/device/riscv_compliance_support/support.c]: I didn't notice this was a regression.

**Review by imphil — APPROVED** [review]
Thanks a lot! Please also force-push to the riscv-compliance repo to retain the changes there.

**Comment by mcy** [comment]
@imphil I can't push to that repository. Someone else should do it.

**Comment by imphil** [comment]
Really? You should have write permissions in https://github.com/lowRISC/riscv-compliance, according to the settings. What error message do you get from GitHub?

**Comment by mcy** [comment]
When I try to push, I get a "please log in" error from git, completely ignoring my SSH key.

**Pull request merged**